### PR TITLE
Add TOSCA YAML to TOSCA XML and TOSCA XML to TOSCA YAML Converter

### PR DESCRIPTION
**Pull request by kleinech**
- Integrate converter into ui and backend
- Change TOSCA model builder to generic builder
- Extend/refactor TOSCA model builder
- Extend/refactor TOSCA YAML model
- Add getGroup to definitions ids
- Update Copyright headers to EPL 2.0
- Remove Christoph Kleine from Copyright Contributors list (EPL 2.0)

Signed-off-by: Christoph Kleine <kleinech.github@gmail.com>

- [ ] Ensure that the commit message is [a good commit message](https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

File: docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The TOSCA YAML files have to be read into a Java model (deserialized) and written from the Java model into files (serialized).
+
+## Considered Alternatives
+
+* [SnakeYAML](https://bitbucket.org/asomov/snakeyaml) writing into intermediate model consisting of default Java types
+* SnakeYAML writing into final Java model
+* Manual reading
+
+## Conclusion
```
Comment (koppor) [inline docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md]: Could you rename that to "Decision Outcome"

We renamed the heading - see https://github.com/eclipse/winery/blob/master/docs/adr/template.md

https://github.com/eclipse/winery/blob/master/docs/adr/template.md

File: docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+* *Chosen Alternative: SnakeYAML writing into intermediate model consisting of default Java types*
+
+# Comparison (Optional)
```
Comment (koppor) [inline docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md]: Please rename to "Pros and Cons of the Alternatives "

File: docs/adr/0011-use-builder-pattern-for-model-classes.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+* Setters, getters and default constructor 
+* Large constructors
+* Factories
+
+## Decision Outcome
+
+* Chosen Alternative: *Builders*
+* https://en.wikipedia.org/wiki/Builder_pattern
+
+## Pros and Cons of the Alternatives <!-- optional -->
```
Comment (koppor) [inline docs/adr/0011-use-builder-pattern-for-model-classes.md]: Please remove `<!-- optional -->`

File: docs/adr/0011-use-builder-pattern-for-model-classes.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+### Builders
+
+* `+` Flexible
+* `+` Simple for complex objects
+* `-` Extensions cause problems () 
+
+### Setters, getters and default constructor 
+
+* `-`
```
Comment (koppor) [inline docs/adr/0011-use-builder-pattern-for-model-classes.md]: Please add points here ore move everything below "Builders" into the "Decision Outcome" section.

File: docs/adr/0011-use-builder-pattern-for-model-classes.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+* Builders
+* Setters, getters and default constructor 
+* Large constructors
+* Factories
+
+## Decision Outcome
+
+* Chosen Alternative: *Builders*
+* https://en.wikipedia.org/wiki/Builder_pattern
```
Comment (koppor) [inline docs/adr/0011-use-builder-pattern-for-model-classes.md]: Please move the link into the "Considered Alternatives" as markdown link: `[Builders](https://en.wikipedia.org/wiki/Builder_pattern)`

File: docs/tosca/yaml/Converter.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
+# TOSCA YAML Converter
+## DataTypes and Property Definition Conversions
+### From YAML to XML
+TOSCA YAML defines `TDataTypes` and T`PropertyDefinition`
```
Comment (koppor) [inline docs/tosca/yaml/Converter.md]: Why is the "T" not contained in the backticks?

File: org.eclipse.winery.common/src/main/java/org/eclipse/winery/common/ids/definitions/DefinitionsChildId.java [hunk 7]
```diff
@@ -85,6 +85,8 @@ public Namespace getNamespace() {
 		return this.namespace;
 	}
 
+	public abstract String getGroup();
```
Comment (koppor) [inline org.eclipse.winery.common/src/main/java/org/eclipse/winery/common/ids/definitions/DefinitionsChildId.java]: Please add a method comment here.

File: org.eclipse.winery.yaml.converter/src/test/resources/yaml/MyTinyToDo/service_template.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      properties:
+        ContainerPort: 80
+        SSHPort:
+        ContainerID:
+        ContainerIP:
+      artifacts:
+        MyTinyToDo_DA:
+          type: DockerContainerArtifact
+          files:
+            - artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/MyTinyToDo_DA/files/tinytodo.zip
```
Comment (koppor) [inline org.eclipse.winery.yaml.converter/src/test/resources/yaml/MyTinyToDo/service_template.yml]: The file `tinytodo.zip` is not contained in this PR. Should it be added or can we use a different scenario?

**Comment by kleinech** [comment]
Changed requested changes

File: docs/adr/0009-manual-tosca-yaml-serialisation.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+The TOSCA YAML files have to be read into a Java model (deserialized) and written from the Java model into files (serialized).
+
+## Considered Alternatives
+
+* Manual serialization
+* [SnakeYAML](https://bitbucket.org/asomov/snakeyaml)
+* [jackson-dataformat-yaml](https://github.com/FasterXML/jackson-dataformat-yaml)
+
+## Conclusion
```
Comment (koppor) [inline docs/adr/0009-manual-tosca-yaml-serialisation.md]: Please rename to `## Decision Outcome`. See https://github.com/adr/madr

File: docs/adr/0009-manual-tosca-yaml-serialisation.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
+# TOSCA YAML serialization deserialization using snakeyaml
```
Comment (koppor) [inline docs/adr/0009-manual-tosca-yaml-serialisation.md]: Please write snakeyaml as "SnakeYAML"

File: docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+## Considered Alternatives
+
+* [SnakeYAML](https://bitbucket.org/asomov/snakeyaml) writing into intermediate model consisting of default Java types
+* SnakeYAML writing into final Java model
+* Manual reading
+
+## Decision Outcome
+
+* *Chosen Alternative: SnakeYAML writing into intermediate model consisting of default Java types*
```
Comment (koppor) [inline docs/adr/0010-tosca-yaml-deserialisation-using-snakeyaml.md]: Please highlight the text after `Chosen Alternative` 😇

File: docs/adr/0011-use-builder-pattern-for-model-classes.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+* Builders
+* Setters, getters and default constructor 
+* Large constructors
+* Factories
+
+## Decision Outcome
+
+* Chosen Alternative: *Builders*
+* [Builders]
```
Comment (koppor) [inline docs/adr/0011-use-builder-pattern-for-model-classes.md]: Please Move that link to line 7.

**Comment by kleinech** [comment]
Updated adrs

**Comment by koppor** [comment]
Filed 7e4b437 as [CQ 14553](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=14553)

**Pull request merged**